### PR TITLE
Archive

### PR DESCRIPTION
**Pull request by TApplencourt**
add the `--archive` option to use the new `bt2 ctf.archive`.

### CODE REVIEW

**Comment by TApplencourt** [comment]
The CI should still pass, as all modif should be guarded by the `--archive`... Will have a look.

File: xprof/xprof.rb.in [hunk 1]
```diff
@@ -550,10 +550,14 @@ def setup_lttng(backends)
     end
 
   end
+
+  exec("lttng enable-rotation --session=#{lttng_session_uuid} --size=50M --timer=60s") if OPTIONS[:archive]
   exec("lttng start #{lttng_session_uuid}")
+  lttng_session_uuid
```
Comment (TApplencourt) [inline xprof/xprof.rb.in:556]:
```suggestion  
```

File: xprof/xprof.rb.in [hunk 2]
```diff
@@ -749,6 +773,7 @@ if __FILE__ == $PROGRAM_NAME
   parser.on('-b', '--backends BACKENDS', Array, "Select which and how backends' need to handled.",
             'Format: backend_name[:backend_level],...',
             default: ['omp:2', 'cl:1', 'ze:1', 'cuda:1', 'hip:1'])
+  parser.on('--[no-]archive', 'Trigger for ardhive support', default: false)
```
Comment (bd4) [inline xprof/xprof.rb.in:776]: typo in description, 'ardhive' -> 'archive'

**Review by bd4 — DISMISSED** [review]

File: xprof/xprof.rb.in (hunk 2 above)
Comment (TApplencourt) [reply xprof/xprof.rb.in]: Thanks!

File: utils/babeltrace_thapi.in [hunk 3]
```diff
@@ -130,6 +130,7 @@ def get_components(names)
   components_classes = {
     'source.ctf.fs' => BT2::BTPlugin.find('ctf').get_source_component_class_by_name('fs'),
     'source.ctf.lttng_live' => BT2::BTPlugin.find('ctf').get_source_component_class_by_name('lttng-live'),
+    'source.ctf.lttng-archive' => BT2::BTPlugin.find('ctf').get_source_component_class_by_name('lttng-archive'),
```
Comment (Kerilk) [inline utils/babeltrace_thapi.in:133]:
```suggestion
    'source.ctf.lttng_archive' => BT2::BTPlugin.find('ctf').get_source_component_class_by_name('lttng-archive'),
```

File: utils/babeltrace_thapi.in [hunk 4]
```diff
@@ -200,6 +201,10 @@ def get_and_add_components(graph, names, l_inputs)
       graph.add(comp, 'source_live',
                 params: { 'inputs' => $options[:inputs],
                           'session-not-found-action' => 'end' })
+    when 'source.ctf.lttng-archive'
```
Comment (Kerilk) [inline utils/babeltrace_thapi.in:204]:
```suggestion
    when 'source.ctf.lttng_archive'
```

File: utils/babeltrace_thapi.in [hunk 5]
```diff
@@ -200,6 +201,10 @@ def get_and_add_components(graph, names, l_inputs)
...
+      graph.add(comp, 'source-archive',
```
Comment (Kerilk) [inline utils/babeltrace_thapi.in:205]:
```suggestion
    when 'source.ctf.lttng_archive'
      graph.add(comp, 'source_archive',
```

File: utils/babeltrace_thapi.in [hunk 6]
```diff
@@ -280,6 +285,8 @@ def bt_graphs
   @bt_graphs ||= begin
     source_comp = if $options[:live]
                     'source.ctf.lttng_live'
+                  elsif $options[:archive]
+                    'source.ctf.lttng-archive'
```
Comment (Kerilk) [inline utils/babeltrace_thapi.in:289]:
```suggestion
                    'source.ctf.lttng_archive'
```

File: xprof/xprof.rb.in [hunk 7]
```diff
@@ -587,15 +588,30 @@ def on_node_processing_babeltrace(backends)
...
+    opts << "--archive #{lttng_session_uuid} --session-found-file-path=#{read_file}"
+    cmd = "#{BINDIR}/babeltrace_thapi #{opts.join(' ')} -- #{lttng_trace_dir_tmp}"
+    LOGGER.debug(cmd)
+    pid_bt = spawn(cmd)
+
+    cmd = "laurence.py --log-level=CRITICAL #{lttng_session_uuid} 500"
+    LOGGER.debug(cmd)
+    pid_laurence = spawn(cmd)
+
+    sleep(1) until File.exist?(read_file)
```
Comment (Kerilk) [inline xprof/xprof.rb.in:603]: Too much?

File: xprof/xprof.rb.in [hunk 8]
```diff
@@ -635,25 +652,32 @@ def trace_and_on_node_processing(usr_argv)
     # All ranks need to set the LLTTNG_HOME env
     # so they can have access to the daemon
     ENV['LTTNG_HOME'] = lttng_home_dir
+    LOGGER.debug("LTTNG_HOME = #{ENV.fetch('LTTNG_HOME', nil)}")
```
Comment (Kerilk) [inline xprof/xprof.rb.in:655]:
```suggestion
```

File: xprof/xprof.rb.in (hunk 7 above)
Comment (TApplencourt) [reply xprof/xprof.rb.in]: Done in 7da4147

**Comment by TApplencourt** [comment]
ready

**Pull request closed without merging**